components/bl-mapbox-component/src/index.js

```diff
@@ -0,0 +1,224 @@
+import { useEffect, useRef, useState } from 'react';
+
+import mapboxgl from './lib/mapbox';
+import MapboxDirections from './lib/mapbox-directions';
+import MapboxGeocoder from './lib/mapbox-geocoder';
+import { createActions } from './actions';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const defaultValues = {
+  START_POS           : { lat: 0, lng: 0 },
+  MAP_STYLE           : 'mapbox://styles/mapbox/streets-v11',
+  ZOOM                : 10,
+  PROJECTION          : 'mercator',
+  LOWER_ATMOSPHERE    : '#BAD2EB',
+  UPPER_ATMOSPHERE    : '#245CDF',
+  ATMOSPHERE_THICKNESS: 0.2,
+  SPACE_COLOR         : '#0B0B19',
+  STAR_INTENSITY      : 0.2,
+};
+
+export default function Mapbox({ component, eventHandlers }) {
+  const mapContainerRef = useRef(null);
+  const mapRef = useRef(null);
+
+  const [polygonsObject, setPolygonsObject] = useState([]);
+
+  const { onClick, onMarkerClick, onPolygonClick, onDeterminingGeoposition, onPan } = eventHandlers;
+
+  const {
+    markers, polygons, center, zoom, fullScreen, accessToken, projection, lowerAtmosphere, upperAtmosphere,
+    atmosphereThickness, spaceColor, starIntensity, mapStyle, navigation, searchBar, geolocation, directions,
+    display, classList,
+  } = component;
+
+  if (!display) {
+    return null;
+  }
+
+  useEffect(() => {
+    if (polygons) {
+      if (polygons.length) {
+        mapRef.current.on('load', () => {
+          polygonsObject.forEach(polygon => {
+            if (mapRef.current.getSource(polygon)) {
+              mapRef.current.removeLayer(`${ polygon.id }-layer`);
+              mapRef.current.removeSource(polygon.id);
+            }
+          });
+        });
+
+        setPolygonsObject(polygons.map(polygon => ({
+          id         : BackendlessUI.UUID.short(),
+          color      : polygon.color,
+          coordinates: polygon.polygon.boundary.points.map(coordinate => [coordinate.lng, coordinate.lat]),
+          name       : polygon.name,
+          opacity    : polygon.opacity,
+          description: polygon.description,
+        })));
```

Comment:
setPolygonsObject - but it's array

```suggestion
        setPolygonsObject(preparePolygons(polygons));
```